cameo/flux_analysis/analysis.py

```diff
@@ -367,6 +379,100 @@ class _PhenotypicPhasePlaneChunkEvaluator(object):
     def __init__(self, model, variable_reactions):
         self.model = model
         self.variable_reactions = variable_reactions
+        objective_reactions = [reaction for reaction in model.reactions
+                               if reaction.objective_coefficient != 0]
+        if len(objective_reactions) != 1:
+            raise Exception('multiple objectives not supported')
+        self.product_reaction = objective_reactions[0]
+        self.c_source_reactions = self.get_c_source_reactions()
+
+    def get_c_source_reactions(self):
+        """ carbon source reactions
+
+        Returns
+        -------
+        list
+           list of media reaction currently with negative flux or 0 if non defined
+        """
+        medium_reactions = [self.model.reactions.get_by_id(reaction) for reaction in self.model.medium.reaction_id]
+        self.model.solve()
+        source_reactions = [reaction for reaction in medium_reactions if
+                            (reaction.n_carbon > 0) and (reaction.flux < 0)]
```

Comment:
True. My concern is parallelization because reaction.flux will fetch directly from the state of the solver. Ok, I agree on keeping *reaction.flux* until we figure out a better implementation. But there is no need to except any kind of SolverError inside *carbon_yield* and *mass_yield*.